### PR TITLE
`std.testing`: make the caret indicator line more helpful

### PR DESCRIPTION
**Pull request by r00ster91**
I think this will make it even easier to debug non-matching strings. Some examples:
```zig
    try testing.expectEqualStrings("hello", "hell");
```
```
====== expected this output: =========
hello␃

======== instead found this: =========
hell␃

======================================
First difference occurs on line 1:
expected:
hello
    ^ (111)
found:
hell
    ^ (out of bounds)
```
```zig
    const not_hello: [5]u8 = undefined;
    try testing.expectEqualStrings("hello", &not_hello);
```
```
====== expected this output: =========
hello␃

======== instead found this: =========
␃

======================================
First difference occurs on line 1:
expected:
hello
^ (104)
found:

^ (0)
```
This is extremely useful if your character is unprintable.
Wasn't sure if this should be hexadecimal so for now I used `{d}`.

### CODE REVIEW

File: lib/std/testing.zig [hunk 1]
```diff
@@ -501,7 +501,10 @@ fn printIndicatorLine(source: []const u8, indicator_index: usize) void {
         while (i < indicator_index) : (i += 1)
             print(" ", .{});
     }
-    print("^\n", .{});
+    if (indicator_index >= source.len)
+        print("^ (out of bounds)\n", .{})
```
Comment (motiejus) [inline lib/std/testing.zig:505]: nit: perhaps call this `(end of string)`?

**Review by motiejus — APPROVED** [review]
I really would have liked this a few weeks ago when I was debugging many strings, some of which had non-printable characters. :) Looking forward.

File: lib/std/testing.zig (hunk 1 above)
Comment (r00ster91) [reply lib/std/testing.zig:505]: That's better I think!

**Comment by r00ster91** [comment]
#12214

**Pull request closed without merging**